### PR TITLE
add Roboto-Condensed and SourceSansPro fonts

### LINKED ISSUE #4: Add Roboto and Souce Sans Pro
**Description**
Apart from CC Accidenz Commons, the CC sites use 'Roboto' and 'Source Sans Pro' in their UIs. Citing benefits listed in https://github.com/creativecommons/fonts/issues/1#issuecomment-548986367, it would be wise to include those fonts here as well.

**Problems solved**
https://github.com/creativecommons/fonts/issues/1#issuecomment-548986367

**Alternatives**
We are currently importing those fonts into Vocabulary from Google Fonts, which works perfectly fine although there is scope for improvement.

### PR DESCRIPTION
**Pull request by Tanuj22**
**Fixes**

Fixes #4 by @dhruvkb 

**Description**

This PR add fonts 'Roboto Condensed' and 'Source Sans Pro'

**Type of PR** 
This PR is a feature.

**Checklist:**
 
- [x] My pull request has a descriptive title (not a vague title like "Update `index.md`").
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

**Legal**
By submitting this PR, I agree to abide by the terms of the Developer 
Certificate of Origin.

<details>
<summary>Developer Certificate of Origin</summary>

Developer Certificate of Origin<br>
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.<br>
1 Letterman Drive<br>
Suite D4700<br>
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
@Tanuj22 WOFF and WOFF2 are the recommended font formats for web usage. OTF and TTF are better suited for local usage.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Add WOFF and WOFF2 fonts, either replacing OTF or using OTF as a fallback.

**Comment by Tanuj22** [comment]
@dhruvkb I have made the requested changes. Please check them out.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Sass has data types like dictionaries and lists and allows us to use iteration to generate CSS. You should use those features to eliminate code duplication from your `.scss` files.

**Comment by Tanuj22** [comment]
@dhruvkb  Please checkout the changes

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Since the unicode range and the loop are very similar for Roboto Condensed and Source Sans Pro, how about making a mixin in `src/styles/font_face.scss` like so

```scss
@mixin font-face($font-name, $font-file) { 
  ...
}
```
and just calling it from both `.scss` files?

**Comment by Tanuj22** [comment]
Since the font weight is different for different Source Sans Pro . It would be something like this right?
`@mixin font-face($font-name, $font-file,$font-weight,$unicode) { 
  ...
}
`

**Comment by Tanuj22** [comment]
And sorry if I am taking a bit long getting done with by PRs. Actually I am traveling currently due to sudden close of college due to coronavirus .

**Comment by Tanuj22** [comment]
@dhruvkb  Please checkout the changes

File: src/styles/source_sans_pro.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
+@import "./font_face.scss";
+
+$unicodes : (U+0460-052F, U+1C80-1C88, U+20B4, U+2DE0-2DFF, U+A640-A69F, U+FE2E-FE2F),
+        (U+0400-045F, U+0490-0491, U+04B0-04B1, U+2116),
+        (U+1F00-1FFF),
+        (U+0370-03FF),
+        (U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1, U+01AF-01B0, U+1EA0-1EF9, U+20AB),
+        (U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF),
+        (U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD);
```
Comment (dhruvkb) [inline src/styles/source_sans_pro.scss]: Can you put this variable in `./font_face.scss`? Since it's the same for both fonts?

**Review by dhruvkb — CHANGES_REQUESTED** [review]
Excellent work on this. Can you quickly incorporate this very minor change?

File: src/styles/source_sans_pro.scss (hunk 1 above)
Comment (Tanuj22) [reply src/styles/source_sans_pro.scss]: Done

**Comment by dhruvkb** [comment]
@Tanuj22 I don't want to delay this any more than I already have, it's a beautiful PR. Can you incorporate the changes from #6 to add a similar tables for the fonts you added here? It'll be slightly more complicated because of the multiple font weights (as compared to only one for CC Accidenz Commons) but that's the last change I'll request.

**Comment by Tanuj22** [comment]
Okay, on it. Should there be different stories for different font weights?

**Comment by dhruvkb** [comment]
I think you can add multiple tables in the same story.

**Comment by Tanuj22** [comment]
@dhruvkb I am not quite sure if this is the correct approach. Please take a look.

**Review by dhruvkb — APPROVED** [review]
Won my heart with the triple use of `{storyFn()}` in the decorator.

**Comment by Tanuj22** [comment]
Thanks! :smiley:

**Pull request merged**